src/ipc/ipc3/dai.c

```diff
@@ -283,17 +284,31 @@ int dai_config(struct comp_dev *dev, struct ipc_config_dai *common_config,
 	comp_info(dev, "dai_config() dai type = %d index = %d dd %p",
 		  config->type, config->dai_index, dd);
 
+	if (dd->chan) {
+		/* perform post trigger DMA ops */
+		switch (config->flags) {
+		case SOF_DAI_CONFIG_FLAGS_HW_FREE:
+			ret = dma_post_stop(dd->chan);
+			if (ret < 0)
+				return ret;
+
+			/* reset DAI config for HDA DAI's */
+			if (ipc_config->type == SOF_DAI_INTEL_HDA)
+				dai_config_reset(dev);
+			break;
+		default:
+			break;
+		}
+
+		return ret;
```

Comment:
you're always returning 0 here, so you can use `return 0` here and avoid initialising `ret`